sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -39,6 +39,24 @@ namespace Azure { namespace Core {
    */
   class Context {
   public:
+    /**
+     * @brief A context key.
+     */
+    struct Key final
```

Comment:
Consider making it non-copiable, non-movable, since copy and move operations will mess up `m_uniqueAddress`

Reply:
I wanted to, but this needs to be default constructible and copyable, so that default instance of Context can be created and be assigned one another.

Reply:
ok

Reply:
should we use `Context::Key&` or `Context::Key*` as member?

Reply:
Reference members don't sound right to me - they are harder to debug. Plus, we compare addresses, so storing pointer is more natural, and foes not require `&` during comparison. Plus, when you have a reference member, the instance is unassignable.

Reply:
Then how about pointers?